crates/ty_python_semantic/resources/mdtest/import/nonstandard_conventions.md

```diff
@@ -1,39 +1,35 @@
 # Nonstandard Import Conventions
 
 This document covers ty-specific extensions to the
-[standard import conventions](https://typing.python.org/en/latest/spec/distributing.html#import-conventions).
-
-It's a common idiom for a package's `__init__.py(i)` to include several imports like
-`from . import mysubmodule`, with the intent that the `mypackage.mysubmodule` attribute should work
-for anyone who only imports `mypackage`.
-
-In the context of a `.py` we handle this well through our general attempts to faithfully implement
-import side-effects. However for `.pyi` files we are expected to apply
-[a more strict set of rules](https://typing.python.org/en/latest/spec/distributing.html#import-conventions)
-to encourage intentional API design. Although `.pyi` files are explicitly designed to work with
-typecheckers, which ostensibly should all enforce these strict rules, every typechecker has its own
-defacto "extensions" to them and so a few idioms like `from . import mysubmodule` have found their
-way into `.pyi` files too.
-
-Thus for the sake of compatibility, we need to define our own "extensions". Any extensions we define
-here have several competing concerns:
-
-- Extensions should ideally be kept narrow to continue to encourage explicit API design
-- Extensions should be easy to explain, document, and understand
-- Extensions should ideally still be a subset of runtime behaviour (if it works in a stub, it works
-    at runtime)
-- Extensions should ideally not make `.pyi` files more permissive than `.py` files (if it works in a
-    stub, it works in an impl)
-
-To that end we define the following extension:
-
-> If an `__init__.pyi` for `mypackage` contains a `from...import` targetting a direct submodule of
-> `mypackage`, then that submodule should be available as an attribute of `mypackage`.
+[standard import conventions](https://typing.python.org/en/latest/spec/distributing.html#import-conventions),
+and other intentional deviations from actual python semantics.
+
+This file currently covers the following details:
+
+- **froms are locals**: a `from..import` can only define locals, it does not have global
+    side-effects. Specifically any submodule attribute `a` that's implicitly introduced by either
+    `from .a import b` or `from . import a as b` (in an `__init__.py(i)`) is a local and not a
+    global. If you do such an import at the top of a file you won't notice this. However if you do
+    such an import in a function, that means it will only be function-scoped (so you'll need to do
+    it in every function that wants to access it, making your code less sensitive to execution
+    order).
```

Comment:
I assume that in the upcoming PR you mentioned in person, this whole paragraph goes away to be replaced by something that says "we only model submodule imports adding the submodule attribute for global-scope submodule imports"